appmap/_implementation/configuration.py

```diff
@@ -0,0 +1,175 @@
+    that should be excluded.
+
+    When checking to see if a function should be included,
+    ConfigFilter looks for the function in both sets.  If the function
+    is found in the included set, it's wrapped for instrumentation.
+    If it's found in the excluded set, it's used without modification.
+    If it isn't found in either set, ConfigFilter passes the function
+    on to the next Filter.
+
+    To determine whether a function is in one of the sets, the
+    fully-qualified name of the function is considered, including the
+    packages(s), module, and class that contain it.  As an example,
+    consider a function named `pkg1.pkg2.mod1.Class1.func`.  Any of
+    `pkg1`, `pkg1.pkg2.`, `pkg1.pkg2.mod1`, `pkg1.pkg2.mod1.Class1`,
+    or `pkg1.pkg2.mod1.Class1.func` will match this function (causing
+    it to be included or excluded, as appropriate).  Note that only
+    full names will match.  (This is achieved in the code by appending
+    `.` to the name retrieved from the config file.)
+    """
+    def __init__(self, *args):
+        super().__init__(*args)
+        if not env.enabled():
+            return
+
+        # Store entries from the config file. Each entry has '.'
+        # appended to it, to ensure that only full names are matched.
+        self.includes = set()
+        self.excludes = set()
+
+        config_file = os.getenv("APPMAP_CONFIG", "appmap.yml")
+        with open(config_file) as file:
+            config = yaml.load(file, Loader=yaml.BaseLoader)
+            logger.debug('config %s', config)
+
+        for package in config['packages']:
+            path = package['path']
+            self.includes.add(path + '.')
+            if 'exclude' in package:
+                excludes = [f'{path}.{e}.' for e in package['exclude']]
+                self.excludes.update(excludes)
+        logger.debug('ConfigFilter, includes %s', self.includes)
+        logger.debug('ConfigFilter, excludes %s', self.excludes)
+
+    def excluded(self, fn):
+        ret = function_in_set(fn, self.excludes)
+        logger.debug('ConfigFilter, %s excluded? %s', fn, ret)
+        return ret
+
+    def included(self, fn):
+        ret = function_in_set(fn, self.includes)
+        logger.debug('ConfigFilter, %s included? %s', fn, ret)
+        return ret
+
+    def filter(self, class_):
+        name = f'{class_.__module__}.{class_.__name__}'
+        logger.debug('ConfigFilter.filter, name %s', name)
+        if in_set(name, self.excludes):
+            logger.debug('  excluded')
+            return False
+        if in_set(name, self.includes):
+            logger.debug('  included')
+            return True
+
+        logger.debug('  undecided')
+        return self.next_filter.filter(class_)
+
+    def wrap(self, fn, isstatic):
+        if self.excluded(fn):
+            return fn
+        if self.included(fn):
+            if getattr(fn, '_appmap_wrapped', None) is None:
+                return wrap(fn, isstatic)
```

Comment:
Should we rename global `def wrap` to make it less confusing since this class method is also the same name.